roles/ci_gen_kustomize_values/templates/nfv/sriov/edpm-values/values.yaml.j2

```diff
@@ -0,0 +1,63 @@
+{% set instances_names = []                                                            %}
+{% set _original_nodeset = (original_content.data | default({})).nodeset | default({}) %}
+{% set _original_nodes = _original_nodeset.nodes | default({})                         %}
+{% set _vm_type = (_original_nodes.keys() | first).split('-')[1]                       %}
+{% for _inst in cifmw_networking_env_definition.instances.keys()                       %}
+{%   if _inst.startswith(_vm_type)                                                     %}
+{%     set _ = instances_names.append(_inst)                                           %}
+{%   endif                                                                             %}
+{% endfor                                                                              %}
+data:
+  preProvisioned: false
+  baremetalSetTemplate:
+{% for instance in instances_names                                                     %}
+    ctlplaneInterface: "{{ cifmw_networking_env_definition.instances[instance].networks.ctlplane.interface_name }}"
+    cloudUserName: cloud-admin
+{%   if cifmw_ci_gen_kustomize_values_provisioning_interface != ''                     %}
+    provisioningInterface: {{ cifmw_ci_gen_kustomize_values_provisioning_interface }}
+{%   endif                                                                             %}
+    bmhLabelSelector:
+      app: openstack
+{% endfor                                                                              %}
+  ssh_keys:
+    authorized: {{ cifmw_ci_gen_kustomize_values_ssh_authorizedkeys | b64encode }}
+    private: {{ cifmw_ci_gen_kustomize_values_ssh_private_key | b64encode }}
+    public: {{ cifmw_ci_gen_kustomize_values_ssh_public_key | b64encode }}
+  nova:
+    migration:
+      ssh_keys:
+        private: {{ cifmw_ci_gen_kustomize_values_migration_priv_key | b64encode }}
+        public: {{ cifmw_ci_gen_kustomize_values_migration_pub_key | b64encode }}
+  nodeset:
+    ansible:
+      ansibleUser: "zuul"
+      ansibleVars:
+        timesync_ntp_servers:
+          - hostname: "{{ cifmw_ci_gen_kustomize_values_ntp_srv | default('pool.ntp.org') }}"
+        edpm_network_config_os_net_config_mappings:
+{% for instance in instances_names                                                     %}
+          edpm-{{ instance }}:
+            nic2: "{{ cifmw_networking_env_definition.instances[instance].networks.ctlplane.mac_addr }}"
+{% endfor                                                                              %}
```

Comment:
I'm wondering if we wouldn't benefit of template snippets..... This loop here is exactly the same we have in HCI - and we're talking right now about injecting all the known NIC (or at least nic1, for sure).

Maybe we should consider something to avoid code duplication?

For the records, it was used in the nftables module nested in edpm-ansible: https://github.com/openstack-k8s-operators/edpm-ansible/blob/main/roles/edpm_nftables/templates/ruleset.j2

Might be interesting in this case...

Reply:
Yep, i want to do a major rewrite of the templates after this patch will be merged. :)